sbi/inference/potentials/likelihood_based_potential.py

```diff
@@ -112,36 +111,23 @@ def _log_likelihoods_over_trials(
 
     Args:
         x: batch of iid data.
-        theta: batch of parameters
-        net: neural net with .log_prob()
+        theta: batch of parameters.
+        estimator: DensityEstimator.
         track_gradients: Whether to track gradients.
 
     Returns:
         log_likelihood_trial_sum: log likelihood for each parameter, summed over all
             batch entries (iid trials) in `x`.
     """
-
-    # Repeat `x` in case of evaluation on multiple `theta`. This is needed below in
-    # when calling nflows in order to have matching shapes of theta and context x
-    # at neural network evaluation time.
-    theta_repeated, x_repeated = match_theta_and_x_batch_shapes(
-        theta=atleast_2d(theta), x=atleast_2d(x)
-    )
-    assert (
-        x_repeated.shape[0] == theta_repeated.shape[0]
-    ), "x and theta must match in batch shape."
-    assert (
```

Comment:
we should keep the device check, no? Or was this moved somewhere else?

Reply:
You're right - I'll add back this assert